pkg/networkserver/downlink.go

```diff
@@ -849,14 +846,26 @@ loop:
 	return phy.DataRates[maxUpDRIdx].DefaultMaxSize.PayloadSize(fp.DwellTime.GetUplinks())
 }
 
+// networkInitiatedDownlinkInterval is the minimum time.Duration passed before a network-initiated(e.g. Class B or C) downlink following an arbitrary downlink.
+const networkInitiatedDownlinkInterval = 5 * time.Second
+
 // downlinkRetryInterval is the time interval, which defines the interval between downlink task retries.
 const downlinkRetryInterval = 2 * time.Second
 
-// gsScheduleWindow is the time interval, which is sufficient for GS to ensure downlink is scheduled.
-const gsScheduleWindow = 30 * time.Second
+// gsScheduleWindow returns minimum time.Duration between downlink being sent to GS for transmission and it being transmitted to the device by the gateway.
```

Comment:
This is longer in peering scenarios.

What is this used for exactly?

What is the consequence of a lower and a higher value?

"returns minimum time" implies that it doesn't get faster than this, but GS can make a gateway transmit within 500 ms

Reply:
So this is the absolute minimum time GS needs to be able to schedule a downlink.
If downlink is to be scheduled at `T`, this is lowest `d`, where downlink sent to GS at `T-d` will be scheduled at 99% of the cases.

Reply:
NS uses this value to select a ping slot to attempt, for example

Reply:
Discussed offline; conclusion is that we're budgeting the RX1 delay and give 1 second and 50% of the remaining to getting back to GS. We cannot make NS aware of latencies of individual downlink paths due to protocol restrictions, complexity and predictability.